packages/drivers/driver-base/src/documentDeltaConnection.ts

```diff
@@ -149,11 +159,31 @@ export class DocumentDeltaConnection
             // though some logic (naming assert in initialMessages getter) might need to be adjusted (it becomes noop)
             assert((this.listeners(event).length !== 0) === this.trackedListeners.has(event), 0x20b /* "mismatch" */);
             if (!this.trackedListeners.has(event)) {
-                this.addTrackedListener(
-                    event,
-                    (...args: any[]) => {
-                        this.emit(event, ...args);
-                    });
+                let listener: (...args: any[]) => void;
+                switch (event) {
+                    case "op": {
+                        listener = (docId: string, messages: ISequencedDocumentMessage[]) => {
+                            if (!this.lightOpAcksEnabled) {
```

Comment:
Not sure if code becomes easier to read, but you can move this check up, i.e.
if (event === "op" && this.lightOpAcksEnabled) {
   // more complicated path
} else {
    listener = (...args: any[]) => { this.emit(event, ...args); };
}